torch/distributed/_shard/sharded_tensor/_ops/math_ops.py

```diff
@@ -120,3 +126,70 @@ def binary_math_op(types, args=(), kwargs=None, pg=None):
 
 for op in binary_ops:
     register_math_op(op)
+
+
+def sharded_bmm_check(*args, **kwargs):
+    """
+    Perform extra checks for the sharded_bmm op, for example, mat2 needs to
+    be a sharded tensor and both tensors need to sharded by dim 0, etc.
+
+    Args: same as ``torch.bmm``.
+
+    Return: None
+    """
+    if len(args) < 2:
+        raise TypeError("Needs two tensors to perform torch.bmm.")
+    st = args[0]
+    mat2 = args[1]
```

Comment:
nit: Maybe st1 and st2 or mat1 and mat2 to be consistent?

Reply:
sure.